packages/bitcore-node/src/modules/ethereum/api/csp.ts

```diff
@@ -134,9 +165,12 @@ export class ETHStateProvider extends InternalStateProvider implements IChainSta
     const { chain, network, address } = params;
     const { web3 } = await this.getWeb3(network);
     const tokenAddress = params.args && params.args.tokenAddress;
+    const multisigContractAddress = params.args && params.args.multisigContractAddress;
     const addressLower = address.toLowerCase();
     const cacheKey = tokenAddress
       ? `getBalanceForAddress-${chain}-${network}-${addressLower}-${tokenAddress.toLowerCase()}`
+      : multisigContractAddress
+      ? `getBalanceForAddress-${chain}-${network}-${multisigContractAddress.toLowerCase()}`
```

Comment:
Thinking we might not need this distinction. On the BWS side we should pass in the multisigContractAddress as the address

Reply:
We are treating multisig like erc20 wallets, so there is not wallet address data stored. 
```
let addresses = await this.getWalletAddresses(params.wallet._id);
```
^ since `params.wallet._id` is the id from the linked eth wallet to that multisig contract, this will get the linked wallet addresses. That's the reason why I need the `multisigContractAddress` variable in the params to get the correct balance.